src/Orchard.Web/Modules/Orchard.Autoroute/Settings/AutorouteSettingsEvents.cs

```diff
@@ -39,31 +94,37 @@ public class AutorouteSettingsHooks : ContentDefinitionEditorEventsBase {
                 Patterns = new List<RoutePattern>()
             };
 
-            if (updateModel.TryUpdateModel(settings, "AutorouteSettings", null, null)) {
+            //get cultures
+            settings.SiteCultures = _cultureManager.ListCultures().ToList();
 
-                var defaultPattern = settings.Patterns[settings.DefaultPatternIndex];
+            if (updateModel.TryUpdateModel(settings, "AutorouteSettings", null, null)) {
+                //TODO need to add validations client and/or server side here
                 // remove empty patterns
                 var patterns = settings.Patterns;
                 patterns.RemoveAll(p => String.IsNullOrWhiteSpace(p.Name) && String.IsNullOrWhiteSpace(p.Pattern) && String.IsNullOrWhiteSpace(p.Description));
 
-                if (patterns.Count == 0) {
-                    patterns.Add(new RoutePattern {
-                        Name = "Title",
-                        Description = "my-title",
-                        Pattern = "{Content.Slug}"
-                    });
-
-                    _notifier.Warning(T("A default pattern has been added to AutoroutePart"));
+                //If there is no default pattern for each culture we set default ones
+                List<RoutePattern> newPatterns = new List<RoutePattern>();
+                int current = 0;
+                foreach (string culture in settings.SiteCultures) {
+                    if (settings.Patterns.Any(x => x.Culture == culture)) {

```

Comment:
I could add this everywhere I can find Culture == culture in that module too. Just in case someone edits the value by hand in the database and he uses lowercase and that we know that they are always like this : fr-CA <-- the bad french ;)